src/App.tsx

```diff
@@ -1,61 +1,148 @@
 import './App.scss';
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  const user = usersFromServer.find((person) => person.id === userId);
+
+  return user || null;
+}
+
+function getMaxId(todos: Todo[]): number {
+  return Math.max(...todos.map((todo) => todo.id)) + 1;
+}
+
+export const initialTodos: Todo[] = [...todosFromServer].map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState<number>(0);
+  const [todos, setNewTodo] = useState(initialTodos);
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserIdValid, setIsUserIdValid] = useState(false);
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+    setIsTitleValid(false);
+    setIsUserIdValid(false);
+  };
+
+  const handleFormSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newTodo: Todo = {
+      id: getMaxId(todos),
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    };
+
+    if (!title.length) {
+      setIsTitleValid(true);
+    }
+
+    if (!userId) {
+      setIsUserIdValid(true);
+    }
```

Comment:
```suggestion
    if (!title.length || !userId) {
      setIsTitleValid(!title.length);
      setIsUserIdValid(!userId);
      
      return;
    }
```

Combine logic, and return from the handler to prevent the next action